telnet.c

```diff
@@ -39,13 +39,21 @@ static ssize_t telnet_read(struct ios_ops *ios, void *buf, size_t count)
 
 static int telnet_set_speed(struct ios_ops *ios, unsigned long speed)
 {
+	unsigned char buf2[14] = {IAC, SB, TELNET_OPTION_COM_PORT_CONTROL, SET_BAUDRATE_CS};
+	size_t offset = 4;
+	int i;
+
+	for (i = 0; i < 4; ++i) {
+		buf2[offset] = (speed >> (24 - 8 * i)) & 0xff;
```

Comment:
You need to switch from host to network order htonl() somewhere in here, or the bytes will be backwards if your not on a big endian arch.

The old code was doing this.

Reply:
The switch is there because I read the MSB byte of speed into the first of the four byte positions.

Also note that the old code might be unsafe as for

    *speedp = htonl(speed);

speedp is not guaranteed to be properly aligned.